lib/cdo/poste.rb

```diff
@@ -110,41 +110,69 @@ def self.unsubscribe(email, hashed_email, params={})
   end
 
   class Template
-    def initialize(body, engine=TextRender::MarkdownEngine)
+    def initialize(path)
+      body = IO.read(path)
+
+      @engine = {
+        '.html' => TextRender::ErbEngine,
+        '.md' => TextRender::MarkdownEngine,
+      }[File.extname(path).downcase]
```

Comment:
What happened to haml, txt, and yml?

Reply:
Oh, I should keep haml in there; but txt and yml aren't actually supported extensions: https://github.com/code-dot-org/code-dot-org/blob/4c2808a6796d98b8d855a1e83a03be827be59054/lib/cdo/poste.rb#L77-L79